### PR TITLE
fix: add github action for upgrade testing

### PR DESCRIPTION
**Pull request by nghuyenthevinh2000**
## Summary of changes

Backport version of #91 

Add github action for upgrade testing

Should only be merged after v2 upgrade for treasury is in place (1.0.6)

## Report of required housekeeping

- [ ] Github issue OR spec proposal link
- [ ] Wrote tests
- [ ] Updated API documentation (client/lcd/swagger-ui/swagger.yaml)
- [ ] Added a relevant changelog entry: clog add [section] [stanza] [message]

----

## (FOR ADMIN) Before merging

- [ ] Added appropriate labels to PR
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" (coding standards)
- [ ] Confirm added tests are consistent with the intended behavior of changes
- [ ] Ensure all tests pass

### CODE REVIEW

**Review by ZaradarBH — APPROVED** [review]
LGTM :)

File: contrib/updates/Dockerfile.cosmovisor [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM golang:1.18-alpine3.17 AS builder
```
Comment (ZaradarBH) [inline contrib/updates/Dockerfile.cosmovisor:1]: We dont really need the "AS builder" alias here since its not really a multi-stage build. :)

**Comment by nghuyenthevinh2000** [comment]
I will change the github actions so that this action will only be invoked if it detects app/upgrades has new folder

**Comment by nghuyenthevinh2000** [comment]
This one is good to go: @inon-man @fragwuerdig can you approve it?

A report on a successful upgrade here with this upgrade workflow invoker: https://github.com/classic-terra/core/pull/101/commits/73b2cceca41e18f6acb838b8bcae1c6abc05099c

**Comment by fragwuerdig** [comment]
IMG_20230217_002222

Does not seem to run. It's running like 0s....

**Comment by inon-man** [comment]
@fragwuerdig 

> IMG_20230217_002222
> 
> Does not seem to run. It's running like 0s....

Due to the diff condition check here: https://github.com/classic-terra/core/blame/gh-v1-upgrade-test/.github/workflows/test.yml#L73

**Review by inon-man — APPROVED** [review]

**Pull request merged**